src/ShallowWrapper.js

```diff
@@ -614,21 +614,23 @@ class ShallowWrapper {
    *
    * @param {String} event
    * @param {Array} args
-   * @returns {ShallowWrapper}
+   * @returns {*}
    */
   simulate(event, ...args) {
     const handler = this.prop(propFromEvent(event));
     if (handler) {
-      withSetStateAllowed(() => {
+      return withSetStateAllowed(() => {
         // TODO(lmr): create/use synthetic events
         // TODO(lmr): emulate React's event propagation
+        let response;
         performBatchedUpdates(this, () => {
-          handler(...args);
+          response = handler(...args);
```

Comment:
I wonder if it would be better to return the equivalent of `Promise.try(() => handler(...args))`, to cleanly encapsulate any exceptions - the only downside being that you couldn't get at the result synchronously. Are there use cases for a synchronous non-Promise return value?